test/unity/toolkits/object_detection/test_object_detector.cxx

```diff
@@ -743,7 +768,333 @@ BOOST_AUTO_TEST_CASE(test_object_detector_auto_split) {
+
+  model.create_iterator_calls_.push_back(create_iterator_impl);
+
+  mock_context->create_augmenter_calls_.push_back(create_augmenter_impl);
+
+  mock_context->create_object_detector_calls_.push_back(
+      create_object_detector_impl);
+  model.create_compute_context_calls_.push_back(create_compute_context_impl);
+
+  auto predict_impl = [](const float_array_map& inputs) {
+    float_array_map result;
+
+    static constexpr size_t OUTPUT_GRID_SIZE = 2;
+    static constexpr size_t NUM_CLASSES = 2;
+    static constexpr size_t NUM_PREDS = NUM_CLASSES + 5;  // 4 for bbox, 1 conf
+
+    const std::vector<std::pair<float, float>> anchor_boxes = {
+        {1.f, 2.f},
+        {1.f, 1.f},
+    };
+
+    // Allocate a YOLO map and define a setter function so we can populate it
+    // relatively conveniently.
+    size_t buffer_size =
+        anchor_boxes.size() * NUM_PREDS * OUTPUT_GRID_SIZE * OUTPUT_GRID_SIZE;
+    std::vector<float> buffer(buffer_size, 0.f);
+    auto set_val = [&](size_t h, size_t w, size_t b, size_t p, float val) {
+      float* out = buffer.data();
+      out += h * NUM_PREDS * anchor_boxes.size() * OUTPUT_GRID_SIZE;
+      out += w * NUM_PREDS * anchor_boxes.size();
+      out += b * NUM_PREDS;
+      out += p;
+      *out = val;
+    };
+
+    // Initialize all confidence scores to large negative values, which
+    // correspond to zero confidence (after passing through the sigmoid
+    // function).
+    for (size_t h = 0; h < OUTPUT_GRID_SIZE; ++h) {
+      for (size_t w = 0; w < OUTPUT_GRID_SIZE; ++w) {
+        for (size_t b = 0; b < anchor_boxes.size(); ++b) {
+          set_val(h, w, b, 4, -1000.f);  // conf
+        }
+      }
+    }
+
+    // Predict class 0 at the center of output cell (1,0) with exactly the size
+    // of anchor box 0.
+    set_val(0, 1, 0, 0, 0.f);     // x
+    set_val(0, 1, 0, 1, 0.f);     // y
+    set_val(0, 1, 0, 2, 0.f);     // w
+    set_val(0, 1, 0, 3, 0.f);     // h
+    set_val(0, 1, 0, 4, 1000.f);  // conf
+    set_val(0, 1, 0, 5, 1000.f);  // class 0
+    set_val(0, 1, 0, 6, 0.f);     // class 1
+
+    // Predict class 1 at the upper-left corner of output cell (0,1) at half the
+    // size of anchor box 1, with confidence 0.5.
+    set_val(1, 0, 1, 0, -1000.f);         // x
+    set_val(1, 0, 1, 1, -1000.f);         // y
+    set_val(1, 0, 1, 2, std::log(0.5f));  // w
+    set_val(1, 0, 1, 3, std::log(0.5f));  // h
+    set_val(1, 0, 1, 4, 0.f);             // conf
+    set_val(1, 0, 1, 5, 0.f);             // class 0
+    set_val(1, 0, 1, 6, 1000.f);          // class 1
+
+    // Create the float_array.
+    shared_float_array arr = shared_float_array::wrap(
+        buffer,
+        {OUTPUT_GRID_SIZE, OUTPUT_GRID_SIZE, anchor_boxes.size() * NUM_PREDS});
+    result["output"] = arr;
+    return result;
+  };
+  mock_nn_model->predict_calls_.push_back(predict_impl);
+
+  auto convert_yolo_impl =
+      [](const neural_net::float_array& yolo_map,
```

Comment:
Given that you went through all the trouble to specify the `yolo_map` returned from the backend (in `predict_impl`), I'm surprised that we don't assert that the `yolo_map` argument here matches. I don't think we need to check everything, but enough to sanity check that the input to `convert_yolo` was the output of `predict`...